rust/chains/hyperlane-ethereum/src/routing_ism.rs

```diff
@@ -0,0 +1,117 @@
+#![allow(clippy::enum_variant_names)]
+#![allow(missing_docs)]
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use ethers::providers::Middleware;
+use tracing::instrument;
+
+use hyperlane_core::{
+    ChainResult, ContractLocator, HyperlaneAbi, HyperlaneChain, HyperlaneContract, HyperlaneDomain,
+    HyperlaneMessage, HyperlaneProvider, RawHyperlaneMessage, RoutingIsm, H256,
+};
+
+use crate::contracts::i_routing_ism::{IRoutingIsm as EthereumRoutingIsmInternal, IROUTINGISM_ABI};
+use crate::trait_builder::BuildableWithProvider;
+use crate::EthereumProvider;
+
+impl<M> std::fmt::Display for EthereumRoutingIsmInternal<M>
```

Comment:
Does this impl need to exist?

Reply:
Not sure, just copied from multisig_ism.rs